pyomo/contrib/mindtpy/util.py

```diff
@@ -963,3 +946,99 @@ def generate_norm_constraint(fp_nlp_model, mip_model, config):
             mip_model.MindtPy_utils.discrete_variable_list,
         ):
             fp_nlp_model.norm_constraint.add(nlp_var - mip_var.value <= rhs)
+
+
+def copy_var_list_values(
+    from_list,
+    to_list,
+    config,
+    skip_stale=False,
+    skip_fixed=True,
+    ignore_integrality=False,
+):
+    """Copy variable values from one list to another.
+    Rounds to Binary/Integer if necessary
+    Sets to zero for NonNegativeReals if necessary
+
+    from_list : list
+        The variables that provides the values to copy from.
+    to_list : list
+        The variables that need to set value.
+    config : ConfigBlock
+        The specific configurations for MindtPy.
+    skip_stale : bool, optional
+        Whether to skip the stale variables, by default False.
+    skip_fixed : bool, optional
+        Whether to skip the fixed variables, by default True.
+    ignore_integrality : bool, optional
+        Whether to ignore the integrality of integer variables, by default False.
+    """
+    for v_from, v_to in zip(from_list, to_list):
+        if skip_stale and v_from.stale:
+            continue  # Skip stale variable values.
+        if skip_fixed and v_to.is_fixed():
+            continue  # Skip fixed variables.
+        var_val = value(v_from, exception=False)
+        set_var_value(
+            v_to,
+            var_val,
+            config.integer_tolerance,
+            config.zero_tolerance,
+            ignore_integrality,
+        )
+
+
+def set_var_value(v_to, var_val, integer_tolerance, zero_tolerance, ignore_integrality):
+    """This function copies variable value from one to another.
+    Rounds to Binary/Integer if necessary.
+    Sets to zero for NonNegativeReals if necessary.
+
+    NOTE: PEP 2180 changes the var behavior so that domain /
+    bounds violations no longer generate exceptions (and
+    instead log warnings).  This means that the following will
+    always succeed and the ValueError should never be raised.
```

Comment:
I think both docstring and plain comments will work.

Reply:
I'm just doesn't feel like documentation to me--it seems more like a comment. It's also not clear what "the following" is referring to anymore since there's no longer a try-except pattern.